src/coreclr/pal/src/thread/process.cpp

```diff
@@ -2567,9 +2567,28 @@ FlushProcessWriteBuffers()
         // Iterate through each of the threads in the list.
         for (mach_msg_type_number_t i = 0; i < cThreads; i++)
         {
-            // Request the threads pointer values to force the thread to emit a memory barrier
-            size_t registers = 128;
-            machret = thread_get_register_pointer_values(pThreads[i], &sp, &registers, registerValues);
+            if (__builtin_available (macOS 10.14, iOS 12, tvOS 9, *))
```

Comment:
Files under src/coreclr/pal are not used by NativeAOT (neither is /debug, and /dlls). I don't see these diffs as problematic - just pointing out that none of this will be exercised anytime soon.

Reply:
Yup, I just wondered since this code is duplicated between here and gcenv.unix.cpp whether it'd be better to keep them in sync. I can remove it if you think that's not worth it - or is there some common helper location we could move them to?

Reply:
We've been trying to slowly move things into src/native/minipal so that everyone can use the same platform abstraction. If this code is leafy enough that we could move it there, we could move it there, but no need to spend too much time on that. The CoreCLR PAL is gnarly which is probably the reason why GC just forked the code.

Reply:
Also cc @janvorli for opinion.

Reply:
I would prefer not touching stuff that's not compiled for NativeAOT.